String reverse

ALeksey5678 opened the pull request.

ALeksey5678 committed:
String reverse

saber0008 commented on src/com/company/reverse/Reverse.java:
```diff
@@ -0,0 +1,21 @@
+        System.out.println(reverse("ad12fgiu57"));
+    }
+
+    private static String reverse(String s) {
```
Please provide more meaningful name for parameter `s`

ALeksey5678 replied:
Done

saber0008 commented on src/com/company/reverse/Reverse.java:
```diff
@@ -0,0 +1,21 @@
+    }
+
+    private static String reverse(String s) {
+        StringBuilder result = new StringBuilder(s.replaceAll("[\\d+]", "")).reverse();
```
Please give it more meaningful name instead of generic `result`

ALeksey5678 replied:
Done

saber0008 commented on src/com/company/reverse/Reverse.java:
```diff
@@ -0,0 +1,21 @@
+    private static String reverse(String s) {
+        StringBuilder result = new StringBuilder(s.replaceAll("[\\d+]", "")).reverse();
+
+        for (int i = 0; i < s.length(); i++) {
```
Please provide more specific name for variable `i`

ALeksey5678 replied:
Done

saber0008 commented on src/com/company/reverse/Reverse.java:
```diff
@@ -0,0 +1,21 @@
+package com.company.reverse;
+
+public class Reverse {
```
It’s suggested that each class should have only one responsibility.
https://en.wikipedia.org/wiki/Single-responsibility_principle

saber0008 replied:
This class actually has at least two responsibilities.
Please decompose it’s logic into multiple classes.

ALeksey5678 replied:
Done

saber0008 reviewed (changes requested):
Looks good so far! 
Left some minor comments about naming mostly.
Also I didn’t actually run your code in IDE so currently no feedback regarding its functionality, will test it later.
Please test some additional cases for example if you input multiple words e.g. `hel1o w0rld`,
Only numbers, input contains special characters and so on

ALeksey5678 committed:
1)name for parameter "s" was change to "string"
2)name instead of generic "result" is "reverseString"
3)renamed variable "i" to "numOfSymbols"
4)decomposing a class is still in progress

saber0008 commented on src/com/company/reverse/Reverse.java:
```diff
@@ -0,0 +1,21 @@
+        System.out.println(reverse("ad12fgiu57"));
+    }
+
+    private static String reverse(String string) {
```
Name `string` Is too generic, and also using type of variable as name considered as bad practice. I suggest to add more context to this name, for example `stringToReverse`, `textToReverse`, `initialText`

ALeksey5678 replied:
Done

saber0008 commented on src/com/company/reverse/Reverse.java:
```diff
@@ -0,0 +1,21 @@
+    private static String reverse(String string) {
+        StringBuilder reverseString = new StringBuilder(string.replaceAll("[\\d+]", "")).reverse();
+
+        for (int numOfSymbols   = 0; numOfSymbols < string.length(); numOfSymbols++) {
```
in this case name `numOfSymbols` Is not so great.
1. It’s shortened (usually we don’t use short words in names, it’s not always obvious what it means) `numberOfSymbols` should work better
2. This name is a bit ambiguous. Judging only by name `numOfSymbols` I will guess that this variable mean some quantity of symbols. But actually it’s more like position or index..

saber0008 replied:
Also formatting is a bit off

ALeksey5678 replied:
Done

saber0008 commented on src/com/company/reverse/Reverse.java:
```diff
@@ -0,0 +1,21 @@
+    }
+
+    private static String reverse(String string) {
+        StringBuilder reverseString = new StringBuilder(string.replaceAll("[\\d+]", "")).reverse();
```
Please put this two string literals into constants

ALeksey5678 replied:
Done

ALeksey5678 committed:
Add ReverseTest class

saber0008 commented on src/com/company/reverse/Reverse.java:
```diff
@@ -0,0 +1,24 @@
+
+public class Reverse {
+
+    static final String ALL_DIGITS_IN_STRING = "[\\d+]";
```
It’s recommended to use most strict access modifiers as possible. If we not intend to expose this field outside we should use `private` modifier

ALeksey5678 replied:
Done

saber0008 commented on src/com/company/reverse/Reverse.java:
```diff
@@ -0,0 +1,24 @@
+public class Reverse {
+
+    static final String ALL_DIGITS_IN_STRING = "[\\d+]";
+    static final String STRING_REGEX = "";
```
The name is too generic

ALeksey5678 replied:
have no idea how to name  ( static final String STRING_REGEX = "" ) properly

saber0008 commented on src/com/company/reverse/Reverse.java:
```diff
@@ -0,0 +1,24 @@
+            }
+        }
+        return reverseString.toString();
+
```
Please clean up empty lines inside methods

ALeksey5678 replied:
Done

saber0008 commented on src/com/company/reverse/ReverseTest.java:
```diff
@@ -0,0 +1,12 @@
+        Reverse reverse = new Reverse();
+
+        System.out.println(reverse.reverse("asdfk124gjh56"));
+
```
Please clean up empty lines

ALeksey5678 replied:
Done

saber0008 commented on src/com/company/reverse/ReverseTest.java:
```diff
@@ -0,0 +1,12 @@
+package com.company.reverse;
+
+public class ReverseTest {
```
Please rename this class to `Application`

ALeksey5678 replied:
Done

saber0008 commented on src/com/company/reverse/Reverse.java:
```diff
@@ -0,0 +1,24 @@
+    static final String ALL_DIGITS_IN_STRING = "[\\d+]";
+    static final String STRING_REGEX = "";
+
+    public String reverse(String stringToReverse) {
```
Please rename either this method or class so trey have different names. It’s a bit confusing

ALeksey5678 replied:
Done

ALeksey5678 committed:
Method reverse was changed to actionOfReversing

saber0008 commented on src/com/company/reverse/Reverse.java:
```diff
@@ -0,0 +1,17 @@
+    private static final String MATCH_ALL_DIGITS = "[\\d+]";
+    private static final String STRING_REGEX = "";
+
+    public String ActionOfReversing(String stringToReverse) {
```
Java style guide suggests different naming strategy for methods. Please see [https://google.github.io/styleguide/javaguide.html#s5.2.3-method-names](url)

ALeksey5678 replied:
Done

ALeksey5678 committed:
Add some changes(rename literal)

ALeksey5678 committed:
rename method name (getReversed)
